### PR TITLE
503s: create DestinationRules before VirtualServices

### PR DESCRIPTION
**Pull request by andraxylia**
Bookinfo changes to ensure when we create and delete VirtualServices via istioctl replace we do not delete the DestinationRules and cause traffic loss.

Associated istio.io change to come.

### CODE REVIEW

File: samples/bookinfo/routing/route-rule-all-v1-mtls.yaml [hunk 1]
```diff
@@ -53,78 +53,3 @@ spec:
         host: details
         subset: v1
 ---
```
Comment (frankbu) [inline samples/bookinfo/routing/route-rule-all-v1-mtls.yaml]: This file is now the same as route-rule-all-v1.yaml. Delete it?

File: samples/bookinfo/routing/destination-rule-all.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+metadata:
+  name: productpage
+spec:
+  host: productpage
+  subsets:
+  - name: v1
+    labels:
+      version: v1
+---
+---
```
Comment (frankbu) [inline samples/bookinfo/routing/destination-rule-all.yaml]: delete this line

File: samples/bookinfo/routing/route-rule-all-v1-mtls.yaml (hunk 1 above)
Comment (andraxylia) [reply samples/bookinfo/routing/route-rule-all-v1-mtls.yaml]: Good catch, removed.

**Comment by andraxylia** [comment]
thanks @rshriram will do

**Comment by andraxylia** [comment]
I notice in fact the mixer test still exercise v1 rule from /kube directory. We need to change them to exercise v1alpha3 rules from the routing directory, cc @douglas-reid .

**Comment by douglas-reid** [comment]
if the tests are running with the old rules, then the flag that was being set for those tests is no longer being set.

This PR made the transition, iirc: https://github.com/istio/istio/pull/6378

**Comment by andraxylia** [comment]
@frankbu can you please approve? All the files there are needed, the destination-rule-all-mtls.yaml contains a different policy.

**Comment by andraxylia** [comment]
@costinm @rshriram @hklai needs admin to be merged.

**Pull request merged**